### ISSUE TITLE
ObjectPage"  ObjectPageSection: current selectedSectionId is not taken into account while render

### ISSUE THREAD

**Issue by star4beam** [SAP/ui5-webcomponents-react#6765]
### Describe the bug

Controlled change of `selectedSectionId` is not taken into account.
Despite the fact that the `selectedSectionId` is changed, the ObjectPage does show the selected section.

Documentation states:
> ... it sets the current selected `ObjectPageSection` by id.
> This can be used to change the selected section by clicking on a custom link or by other user interactions.

### Isolated Example

https://stackblitz.com/edit/github-e1km2ma4-qmfewvz4?file=src%2FApp.tsx,package.json

### Reproduction steps

1. Manually store the selectedSectionId in the React state
2. Change the selectedSectionId interactively (by clicking on a button, etc)
3. The active ObjectPageSection is not changed, the related tab is not highlighted as well.

### Expected Behaviour

The ObjectPage should show the selected section and highlight the related tab when selectedSectionId is changed.
In other words, a controlled way of changing tab should work.

### Screenshots or Videos

<img width="434" alt="Image" src="https://github.com/user-attachments/assets/2c64dcb1-1bed-47be-bc7d-f5d0c109a093" />

### UI5 Web Components for React Version

2.5.1, 1.29.11

### UI5 Web Components Version

2.5.1, 1.24.11

### Browser

Chrome

### Relevant log output

```shell

```

### Organization

SAP

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by gparlakov** [comment]
@Lukas742 is there any chance this gets down-ported to 1.29.x

**Comment by Lukas742** [comment]
Hi @gparlakov 

we will try to add a fix for v1 as well. Unfortunately, we can't simply downport it, as the codebases of the v1 and v2 ObjectPage implementations differ too much, so we’ll need to provide a manual fix, which could take some time.
